jakarta.jdt/org.eclipse.lsp4jakarta.jdt.core/src/main/java/org/eclipse/lsp4jakarta/jdt/core/websocket/WebSocketDiagnosticsCollector.java

```diff
@@ -204,6 +198,45 @@ private void uriMismatchWarningCheck(IType type, List<Diagnostic> diagnostics, I
         }
     }
 
+    /**
+     * Create an error diagnostic if a ServerEndpoint annotation does not follow the rules.
+     */
+    private void serverEndpointErrorCheck(IType type, List<Diagnostic> diagnostics, ICompilationUnit unit) throws JavaModelException {
+        for (IAnnotation annotation : type.getAnnotations()) {
+            if (annotation.getElementName().equals(WebSocketConstants.SERVER_ENDPOINT_ANNOTATION)) {
+                for (IMemberValuePair annotationMemberValuePair : annotation.getMemberValuePairs()) {
+                    if (annotationMemberValuePair.getMemberName().equals(WebSocketConstants.ANNOTATION_VALUE)) {
+                        String path = annotationMemberValuePair
+                                .getValue()
+                                .toString();
+                        Diagnostic diagnostic;
+                        if (!JDTUtils.hasLeadingSlash(path)) {
+                            diagnostic = createDiagnostic(annotation, unit,
+                                    WebSocketConstants.DIAGNOSTIC_SERVER_ENDPOINT_NO_SLASH,
+                                    WebSocketConstants.DIAGNOSTIC_SERVER_ENDPOINT);
+                            diagnostics.add(diagnostic);
+                        } else if (hasRelativePathURIs(path)) {
```

Comment:
I originally had it as simple ifs but with my testing I never got more than one diagnostic to show up so I ended up switching it to else if. I have no problem switching it back maybe there's an edge case I didn't try (like the one you pointed out).

Reply:
Just tried the new changes with your test case and some variations for it. I still don't get more than one diagnostic to show up at a time and any `//`, `/./`, or `/../` sequences don't trigger their respective diagnostic message but show the URI template one instead. Do you want me and go ahead with the new changes?

Reply:
When I ran it I saw strange exceptions in the console from org.eclipse.lsp4e, the framework. When I stepped through your code in the debugger it was working properly and I did see all the messages in the UI:
<img width="627" alt="image" src="https://user-images.githubusercontent.com/24705144/160022808-c5528bf1-51dc-4471-9db9-c0b5a69a27e0.png">
I think the failures you are seeing could be caused by bugs in the framework. 
I find the relative path URI message appears but that the Level 1 URI message also always appears. That makes sense because `//` is going to fail both checks. For that reason it makes sense to test the relative path URI first because it is more specific and use `else if` before the Level 1 URI check so it is only run if the relative path check is ok. The other checks don't interfere with each other. 
It is usually best to flag all the errors at once so the person can fix them all at once. I think your code would be more correct using `else if` just in the one case. 
You could investigate issues against LSP4E to see if any of them match this. Possibly open an issue if you could make a simple test case that demonstrates it. Even investigate the cause.
@kathrynkodama  what do you think? Just generate one message to work around a possible issue in the framework or generate multiple (if required) and hope the framework gets fixed in the future?

Reply:
If I am understanding the above correctly, there are currently multiple diagnostics being shown on the same line (the multiple markers in that screenshot above). 
I would think that only the most relevant diagnostic message is displayed to the user.

Reply:
So keep it at one diagnostic message max with `else if`?

Reply:
Re-reading the above again and I may have misunderstood. The multiple diagnostics appear when there is more than one relevant diagnostic message correct? In that case, I agree with Paul:
> It is usually best to flag all the errors at once so the person can fix them all at once. I think your code would be more correct using else if just in the one case.
This makes sense the most sense to me. If there are issues in the framework when implementing this solution let's open a follow-up bug.